src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java

```diff
@@ -126,36 +127,54 @@ private List<DayOfWeek> normalizeAndSortDayOfWeek() {
   private Date scheduleTaskWithHourPeriod(Date lastActualExecutionTime, Integer hours) {
-    ZonedDateTime startTimeUTC = ScheduleDateTimeUtil.convertScheduleTime(lastActualExecutionTime, scheduleParameters);
-    ZoneId zoneId = ZoneId.of("UTC");
-    ZonedDateTime nowDate = Instant.now().atZone(zoneId);
+    ZonedDateTime startTime = ScheduleDateTimeUtil.convertScheduleTime(lastActualExecutionTime, scheduleParameters);
+
     if (lastActualExecutionTime != null) {
-      long diffHours = (nowDate.toInstant().toEpochMilli() - startTimeUTC.toInstant().toEpochMilli())/(60 * 60 * 1000);
+      ZoneId zoneId = ZoneId.of("UTC");
+      ZonedDateTime nowDate = Instant.now().atZone(zoneId);
+      long diffHours = (nowDate.toInstant().toEpochMilli() - startTime.toInstant().toEpochMilli())/(60 * 60 * 1000);
       if (diffHours > 0 && hours !=0 && diffHours > hours) {
         BigDecimal hoursToIncrease = BigDecimal.valueOf(diffHours)
           .divide(BigDecimal.valueOf(hours), RoundingMode.FLOOR)
           .add(BigDecimal.ONE);
-        startTimeUTC = startTimeUTC.plusHours(hoursToIncrease.longValue() * hours);
-      } else
-      {
-        startTimeUTC = startTimeUTC.plusHours(hours);
+        startTime = startTime.plusHours(hoursToIncrease.longValue() * hours);
+      } else {
+        startTime = startTime.plusHours(hours);
       }
     }
-    log.info("Hourly next schedule execution time in UTC for config {} is : {}", scheduleParameters.getId(), startTimeUTC);
-    return ScheduleDateTimeUtil.convertToOldDateFormat(startTimeUTC, scheduleParameters);
+
+    startTime = normalizeIfNextRunInPast(startTime, getNowDateTime(scheduleParameters.getTimeZone()));
+
+    log.info("Hourly next schedule execution time in {} for config {} is : {}", scheduleParameters.getTimeZone(), scheduleParameters.getId(), startTime);
+    return ScheduleDateTimeUtil.convertToOldDateFormat(startTime, scheduleParameters);
   }
 
   @SneakyThrows
   private Date scheduleTaskWithDayPeriod(Date lastActualExecutionTime, Integer days) {
-    ZonedDateTime startTimeUTC = ScheduleDateTimeUtil.convertScheduleTime(lastActualExecutionTime, scheduleParameters);
+    ZonedDateTime startTime = ScheduleDateTimeUtil.convertScheduleTime(lastActualExecutionTime, scheduleParameters);
     if (lastActualExecutionTime != null) {
-      startTimeUTC = startTimeUTC.plusDays(days);
+      startTime = startTime.plusDays(days);
     }
-    log.info("Day next schedule execution time in UTC for config {} is : {}", scheduleParameters.getId(), startTimeUTC);
-    return ScheduleDateTimeUtil.convertToOldDateFormat(startTimeUTC, scheduleParameters);
+
+    log.info("Day next schedule execution time in {} for config {} is : {}", scheduleParameters.getTimeZone(), scheduleParameters.getId(), startTime);
+    return ScheduleDateTimeUtil.convertToOldDateFormat(startTime, scheduleParameters);
   }
 
+  private ZonedDateTime normalizeIfNextRunInPast(ZonedDateTime startTime, ZonedDateTime nowTime) {
```

Comment:
Some examples for this new logic:
For example, user selects 13.30 as a start time and current time is 16.10 - next run will be at 16.30
For example, user selects 13.10 as a start time and current time is 16.40 - next run will be at 17.10

Reply:
would have been good if this comment was in javadoc. makes it easier to follow the method.

Reply:
Added java doc in currently opened PR: https://github.com/folio-org/mod-data-export-spring/pull/207/files#diff-b9c331f89497219d6d88536ff45eded1984148cd4bc814f9fdd0f4faf2639890R163